services/nodeping/nodeping-uptime.service.js

```diff
@@ -0,0 +1,122 @@
+'use strict'
+
+const { BaseJsonService } = require('..')
+const Joi = require('joi')
+
+const rowSchema = Joi.object().keys({
+  uptime: Joi.number()
+    .precision(3)
+    .min(0)
+    .max(100),
+})
+
+const schema = Joi.array()
+  .items(rowSchema)
+  .min(1)
+
+/*
+ * this is the checkUuid for the NodePing.com (as used on the [example page](https://nodeping.com/reporting.html#results))
+ */
+const sampleCheckUuid = 'jkiwn052-ntpp-4lbb-8d45-ihew6d9ucoei'
+
+// TODO: support for custom # of days
+// TODO: support for custom color thresholds
+// TODO: support for custom colors
+// TODO: support for custom '100%' label
+// TODO: support for custom # of decimal places
+
+class NodePingUptime extends BaseJsonService {
+  static get category() {
+    return 'monitoring'
+  }
+
+  static get defaultBadgeData() {
+    return {
+      label: 'Uptime',
+    }
+  }
+
+  static get route() {
+    return {
+      base: 'nodeping/uptime',
+      pattern: ':checkUuid',
+      queryParams: ['label'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'NodePing uptime',
+        namedParams: {
+          checkUuid: sampleCheckUuid,
+        },
+        staticPreview: this.render({ uptime: 99.999 }),
+      },
+      {
+        title: 'NodePing uptime (customized)',
+        namedParams: {
+          checkUuid: sampleCheckUuid,
+        },
+        queryParams: {
+          label: 'Online',
+        },
+        staticPreview: this.render({ uptime: 100, label: 'Online' }),
+      },
+    ]
+  }
+
+  async fetch({ checkUuid }) {
+    const thirtyDaysAgo = new Date(
+      new Date().getTime() - 30 * 24 * 60 * 60 * 1000
+    )
+      .toISOString()
+      .slice(0, 10)
+
+    const rows = await this._requestJson({
+      schema,
+      url: `https://nodeping.com/reports/uptime/${checkUuid}`,
+      options: {
+        method: 'GET',
```

Comment:
When using the `_requestJson` helper you don't need to explicitly specify the verb unless you need something other than `GET`